### PR TITLE
[Fix] Refine tutorials docs

### PR DESCRIPTION
**Pull request by wHao-Wu**
* Fix some urls in tutorials with fixed codebase version
* Fix some typo errors in tutorials

### CODE REVIEW

File: docs/tutorials/config.md [hunk 1]
```diff
@@ -24,15 +24,15 @@ Please refer to [mmcv](https://mmcv.readthedocs.io/en/latest/utils.html#config)
 We follow the below style to name config files. Contributors are advised to follow the same style.
 
 ```
-{model}_[model setting]_{backbone}_{neck}_[norm setting]_[misc]_[batch_per_gpu x gpu]_{schedule}_{dataset}
+{model}_[model setting]_{backbone}_[neck]_[norm setting]_[misc]_[batch_per_gpu x gpu]_{schedule}_{dataset}
```
Comment (wHao-Wu) [inline docs/tutorials/config.md:27]: I'm not sure whether the neck is the required item when naming config files because some models have not neck part.

File: docs/tutorials/config.md [hunk 2]
```diff
@@ -383,14 +383,14 @@ evaluation = dict(pipeline=[  # Pipeline is passed by eval_pipeline created befo
 ])
 lr = 0.008  # Learning rate of optimizers
 optimizer = dict(  # Config used to build optimizer, support all the optimizers in PyTorch whose arguments are also the same as those in PyTorch
-    type='Adam',  # Type of optimizers,   # Type of optimizers, refer to https://github.com/open-mmlab/mmdetection/blob/master/mmdet/core/optimizer/default_constructor.py#L13 for more details
+    type='Adam',  # Type of optimizers,   # Type of optimizers, refer to https://github.com/open-mmlab/mmcv/blob/v1.3.7/mmcv/runner/optimizer/default_constructor.py#L12 for more details
```
Comment (wHao-Wu) [inline docs/tutorials/config.md:386]: Updating the urls with fixed codebase version to avoid the line linking confusion caused by continually updated master branch.

File: docs/tutorials/config.md [hunk 3]
```diff
@@ -417,7 +417,7 @@ gpu_ids = range(0, 1)  # ids of gpus
 Sometimes, you may set `_delete_=True` to ignore some of fields in base configs.
 You may refer to [mmcv](https://mmcv.readthedocs.io/en/latest/utils.html#inherit-from-base-config-with-ignored-fields) for simple illustration.
 
-In MMDetection or MMDetection3D, for example, to change the FPN neck of PointPillars with the following config.
+In MMDetection3D, for example, to change the FPN neck of PointPillars with the following config.
```
Comment (wHao-Wu) [inline docs/tutorials/config.md:420]: Removing "MMDetection" because of the lack of PointPillars in "MMDetection".

File: docs/tutorials/customize_runtime.md [hunk 4]
```diff
@@ -108,7 +108,7 @@ class MyOptimizerConstructor(object):
 
 ```
 
-The default optimizer constructor is implemented [here](https://github.com/open-mmlab/mmcv/blob/9ecd6b0d5ff9d2172c49a182eaa669e9f27bb8e7/mmcv/runner/optimizer/default_constructor.py#L11), which could also serve as a template for new optimizer constructor.
```
Comment (wHao-Wu) [inline docs/tutorials/customize_runtime.md:111]: The version of mmcv is out-of-date. So I updated it to the newest version (with a fixed version instead of the master branch to avoid the line linking confusion).

File: docs/tutorials/data_pipeline.md [hunk 5]
```diff
@@ -15,7 +15,7 @@ defines how to process the annotations and a data pipeline defines all the steps
 A pipeline consists of a sequence of operations. Each operation takes a dict as input and also output a dict for the next transform.
 
 We present a classical pipeline in the following figure. The blue blocks are pipeline operations. With the pipeline going on, each operator can add new keys (marked as green) to the result dict or update the existing keys (marked as orange).
-![pipeline figure](../../resources/data_pipeline.png)
+![](../../resources/data_pipeline.png)
```
Comment (wHao-Wu) [inline docs/tutorials/data_pipeline.md:18]: I found that the annotation to the figure will force show in `readthedocs`. So I remove the annotation (with no side-effect to the original figure).

File: docs/tutorials/waymo.md [hunk 6]
```diff
@@ -101,7 +101,7 @@ Considering there are many similar frames in the original dataset, we can basica
 
 ## Evaluation
 
-For evaluation on waymo, please follow the [instruction](https://github.com/waymo-research/waymo-open-dataset/blob/master/docs/quick_start.md) to build the binary file `compute_detection_metrics_main` for metrics computation and put it into `mmdet3d/core/evaluation/waymo_utils/`.  Basically, you can follow the commands below to install bazel and build the file.
+For evaluation on waymo, please follow the [instruction](https://github.com/waymo-research/waymo-open-dataset/blob/master/docs/quick_start.md/) to build the binary file `compute_detection_metrics_main` for metrics computation and put it into `mmdet3d/core/evaluation/waymo_utils/`.  Basically, you can follow the commands below to install bazel and build the file.
```
Comment (wHao-Wu) [inline docs/tutorials/waymo.md:104]: Ensuring the link is useful in markdown files.

File: docs/tutorials/config.md (hunk 1 above)
Comment (Tai-Wang) [reply docs/tutorials/config.md:27]: Making it optional should be ok

File: docs/tutorials/customize_models.md [hunk 7]
```diff
@@ -170,14 +170,17 @@ custom_imports = dict(
 
 to the config file and avoid modifying the original code.
 
-#### 3. Modify the config file
+#### 3. Use the neck in your config file
```
Comment (Tai-Wang) [inline docs/tutorials/customize_models.md:173]: I find L59 also has typos, should be 'Use the voxel encoder in your config file'. Please also check others.

File: docs/tutorials/customize_runtime.md [hunk 8]
```diff
@@ -54,10 +54,10 @@ __all__ = ['MyOptimizer']
 You also need to import `optimizer` in `mmdet3d/core/__init__.py` by adding:
 
 ```python
-from .optimizer import *  # noqa: F401, F403
+from .optimizer import *
 ```
 
-- Use `custom_imports` in the config to manually import it
+or using `custom_imports` in the config to manually import it
```
Comment (Tai-Wang) [inline docs/tutorials/customize_runtime.md:60]: or use

File: docs/tutorials/data_pipeline.md [hunk 9]
```diff
@@ -98,11 +98,11 @@ For each operation, we list the related dict fields that are added/updated/remov
 
 `GlobalRotScaleTrans`
 - add: pcd_trans, pcd_rotation, pcd_scale_factor
-- update: points, *bbox3d_fields
+- update: points, bbox3d_fields
```
Comment (Wuziyi616) [inline docs/tutorials/data_pipeline.md:101]: I think the `*` here indicate that, `input_dict['bbox3d_fields']` itself is not changed, but `input_dict[item] for item in bbox3d_fields` will be changed. For example [here](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/datasets/pipelines/transforms_3d.py#L607-L608) in `GlobalRotScaleTrans`, it changes all the boxes, where the boxes key in the dict is indicated by `bbox3d_fields`.

**Review by Wuziyi616 — APPROVED** [review]
Almost OK I think.

File: docs/tutorials/data_pipeline.md (hunk 9 above)
Comment (wHao-Wu) [reply docs/tutorials/data_pipeline.md:101]: Yep, I have also considered that case. I just wonder whether to unify the format of `bbox3d_fields` in [data loading](https://github.com/open-mmlab/mmdetection3d/blob/master/docs/tutorials/data_pipeline.md#data-loading) and [pre-processing](https://github.com/open-mmlab/mmdetection3d/blob/master/docs/tutorials/data_pipeline.md#pre-processing), or just to follow the reasonable pipeline in practice.

Comment (Wuziyi616) [reply docs/tutorials/data_pipeline.md:101]: I got your point. But at least `bbox3d_fields` itself is not changed right? I prefer leaving `*bbox3d_fields` unchanged here.

Comment (wHao-Wu) [reply docs/tutorials/data_pipeline.md:101]: You are definitely right. Maybe we need a further review.

File: docs/tutorials/customize_runtime.md [hunk 10]
```diff
@@ -54,10 +54,10 @@ __all__ = ['MyOptimizer']
 You also need to import `optimizer` in `mmdet3d/core/__init__.py` by adding:
 
 ```python
-from .optimizer import *  # noqa: F401, F403
+from .optimizer import *
 ```
 
-- Use `custom_imports` in the config to manually import it
+or use `custom_imports` in the config to manually import it
```
Comment (Tai-Wang) [inline docs/tutorials/customize_runtime.md:60]: - Or use

File: docs/tutorials/customize_runtime.md [hunk 11]
```diff
@@ -248,7 +248,7 @@ __all__ = [..., 'MyHook']
 
 ```
 
-- Use `custom_imports` in the config to manually import it
+or use `custom_imports` in the config to manually import it
```
Comment (Tai-Wang) [inline docs/tutorials/customize_runtime.md:251]: \- Or use

File: docs/tutorials/data_pipeline.md (hunk 9 above)
Comment (Tai-Wang) [reply docs/tutorials/data_pipeline.md:101]: I think keep * will be better.

**Review by Tai-Wang — APPROVED** [review]
Just fixing the comments will be ok

**Review by ZwwWayne — APPROVED** [review]
